cmd/medius/images/push.go

```diff
@@ -186,7 +186,7 @@ func (b *buildAndPublish) handleMetadataError(imageName string, err error) error
 }
 
 func (b *buildAndPublish) getArtifact(artifactInfo *api.ArtifactDetails) (string, error) {
-	artifactReader, err := b.Getter.GetWithChecksumAndContext(b.Ctx, artifactInfo.DownloadURL)
+	artifactReader, err := b.Getter.GetWithChecksumAndContext(b.Ctx, artifactInfo.DownloadURL, artifactInfo.ChecksumHash())
```

Comment:
Why not pass the `func() hash.Hash` down to where it is actually being used?

Reply:
Done.